pkg/vsphere/tasks/waiter.go

```diff
@@ -164,11 +201,11 @@ func logFault(op trace.Operation, fault types.BaseMethodFault) {
 }
 
 func logSoapFault(op trace.Operation, fault types.AnyType) {
-	op.Debugf("unexpected soap fault on task retry : %#v", fault)
+	op.Debugf("unexpected soap fault on task retry : %s", fault)
```

Comment:
The purposes of the tasks package were:
1. retry in the case of temporary errors, such as InProgress
2. operate in a queue aware fashion (such as throttling)
3. provide easy access to error types and messages

We can start adding code to extract the messages from the embedded faults and print that. Dumping the structs is massively verbose and hard to parse.

Also of note is that we get this even for _expected_ errors because we have no way of indicating whether an error is expected or not.